utilities/dashboard/components/run_charts/lineplot.py

```diff
@@ -0,0 +1,321 @@
+    """Add HoverTool to main plot."""
+    bg_var_name = col[-1]
+    var_name = col[0]
+
+    TOOLTIPS = [
+        (nice_names_dict.get(var_name), "@y"),
+        ("Date of survey", "@x{%d-%m-%Y}"),
+        (nice_names_dict.get(bg_var_name), "@cat"),
+    ]
+
+    kwargs = {"tooltips": TOOLTIPS[:-1]} if col[1] is None else {"tooltips": TOOLTIPS}
+
+    p.add_tools(
+        HoverTool(renderers=[renderers], **kwargs, formatters={"@x": "datetime"},)
+    )
+
+    return p
+
+
+def _apply_styling(p):
+    """Declutter main plot."""
+    # grid styling
+    p.xgrid.grid_line_color = None
+    p.ygrid.grid_line_alpha = 0.5
+    p.outline_line_color = None
+
+    # axis styling
+    p.xaxis.formatter = DatetimeTickFormatter(months=["%B %Y"])
+
+    p.yaxis.axis_line_color = None
+    p.yaxis.axis_label_text_font_style = "normal"
+    p.yaxis.axis_label_standoff = 30
+
+    p.axis.major_tick_line_color = None
+    p.axis.minor_tick_line_color = None
+
+    # title and labels
+    p.title.text_font_size = "15pt"
+
+    p.xaxis.axis_label_text_font_size = "12pt"
+    p.yaxis.axis_label_text_font_size = "12pt"
+
+    p.xaxis.major_label_text_font_size = "11pt"
+    p.yaxis.major_label_text_font_size = "11pt"
+
+    # borders
+    p.min_border_left = 50
+    p.min_border_right = 50
+    p.min_border_top = 20
+    p.min_border_bottom = 50
+
+    return p
+
+
+def update_plot(plot, selectors, bounds, variable, bg_var, nice_names_dict):
+    """Activate and de-activate the lines according to variable and bg_var
+
+    Args:
+        plot (bokeh.figure): The plot that will be updated.
+        selectors (dict): Maps tuples of (variable, bg_var) to a a list
+            of active lines
+        bounds (dict): Maps a combination of variable and bounds needed to
+            adjust the plot's y-axis.
+        variable (str): Name of the selected variable (this corresponds
+            to a key in the data dictionary).
+        bg_var (str): Name of the background variable. Only used to select
+            lines.
+        nice_names_dict (dict): Dictionary mapping variables to nice names.
+
+    Returns:
+        bokeh.figure
+
+    """
+    for line in plot.renderers:
+        line.visible = False
+
+    legend_items = []
+    color_iterator = _get_color_iterator()
+
+    for sel, color in zip(selectors[(variable, bg_var)], color_iterator):
+        name = "-".join(str(i) for i in (*sel, bg_var))
+        lines = plot.select({"name": name})
+
+        lines.glyph.line_color = color
+        lines.visible = True
+
+        # store legend items
+        if bg_var is not None:
+            cat = nice_names_dict.get(f"{bg_var}_{sel[1]}")
+            item = (cat, lines)
+            legend_items.append(item)
+
+    _add_legend(plot, legend_items)
+
+    # when necessary, update y-axis label and y-axis range
+    if plot.yaxis.axis_label != nice_names_dict[variable]:
+
+        _update_yaxis(plot, bounds, variable, nice_names_dict)
+
+    return plot
```

Comment:
See what happens if you don't return the plot here and don't change the layout in `create_component.py:update_{}_variable` (lines 123 and 137). Since you select and deselect the lines in place, there should be no reason to change the layout. I think this might be what causes the callbacks to stop working after the first time.